src/billing_handler.rs

```diff
@@ -37,47 +78,60 @@ pub async fn sign_data(appstate: Data<AppState>, data: Json<SigningData>) -> imp
             .body(format!("Error decoding nonce into 32 bytes: {}", err));
     }
 
-    let mut signed_data: Vec<u8> = bytes32_nonce.to_vec();
-    for txhash in signing_data.txhashes {
-        if let Some(cost) = appstate.execution_costs.lock().unwrap().remove(&txhash) {
-            let mut bytes32_txhash = [0u8; 32];
-            if let Err(err) = hex::decode_to_slice(&txhash[2..], &mut bytes32_txhash) {
-                return HttpResponse::InternalServerError().body(format!(
-                    "Error decoding transaction hash into 32 bytes: {}",
-                    err
-                ));
+    let mut bill_claim_data = bytes32_nonce.to_vec();
+    for tx_hash in signing_data.tx_hashes {
+        if let Some(cost) = appstate.execution_costs.lock().unwrap().remove(&tx_hash) {
+            let mut bytes32_tx_hash = [0u8; 32];
+            if let Err(_) = hex::decode_to_slice(&tx_hash[2..], &mut bytes32_tx_hash) {
+                continue;
             }
 
-            signed_data.append(&mut bytes32_txhash.to_vec());
-            signed_data.append(&mut cost.to_be_bytes().to_vec());
-        } else {
-            return HttpResponse::BadRequest().body(format!(
-                "{} tx hash doesn't exist in the current bill",
-                txhash
-            ));
+            bill_claim_data.append(&mut bytes32_tx_hash.to_vec());
+            bill_claim_data.append(&mut cost.to_be_bytes().to_vec());
         }
     }
 
-    let mut hasher = Keccak::v256();
-    hasher.update(&signed_data);
+    if bill_claim_data.len() == 32 {
+        return HttpResponse::BadRequest().body("No tx present in the bill");
```

Comment:
Make it more verbose. `Given transaction hashes are not present in billing data`